lib/facter/util/facts/unit_converter.rb

```diff
@@ -17,10 +17,11 @@ def hertz_to_human_readable(speed)
             speed = speed.to_i
             return if !speed || speed.zero?
 
-            prefix = { 3 => 'k', 6 => 'M', 9 => 'G', 12 => 'T' }
+            prefix = { 0 => '', 3 => 'k', 6 => 'M', 9 => 'G', 12 => 'T' }
             power = Math.log10(speed).floor
-            validated_speed = power.zero? ? speed.to_f : speed.fdiv(10**power)
-            format('%<displayed_speed>.2f', displayed_speed: validated_speed).to_s + ' ' + prefix[power] + 'Hz'
+            display_power = power - power % 3
+            validated_speed = power.zero? ? speed.to_f : speed.fdiv(10**display_power)
```

Comment:
You could extract this into a private method:
```ruby
def compose_speed_data(speed)
  power = Math.log10(speed).floor
  display_power = power - power % 3
  validated_speed = power.zero? ? speed.to_f : speed.fdiv(10**display_power)
  [display_power, validated_speed]
end
```

and it will be used here as:
```ruby
display_power, validated_speed = compose_speed_data(speed)
format('%<displayed_speed>.2f', displayed_speed: validated_speed).to_s + ' ' + prefix[display_power] + 'Hz'
```

This way, the changes will be minimal and rubocop will be happy

Reply:
I'll try something in that direction.

Since I noticed that I also messed up the commit message (wrong issue is referenced), I'll open a second PR. (Alternatively I could force-push to this PR.)

Reply:
I amended the wrong commit message and force-pushed. @gimmyxd I didn't take your suggestion literally, but decided to move the computation of the metric prefix to the new private method, too. I hope that makes sense.